Feat/header-nandhana

nandhanaselvam opened the pull request:
Completed header, now with internationalization

tiramisuuuuuuu committed:
Made a functioning dropdown

Created the functional aspects of the dropdown: when the v button is clicked, a table of options is created; the clicking of different languages is saved into a state variable; and the component that shows the selected language gets rerendered based on the state variable. Also, I saved the state variables into a context that hopefully allows other components to be able to reference the language selection : )

tiramisuuuuuuu committed:
Moved the LanguageContext to a higher page

In the previous commit, I created a context to store the "language" state variable. In this commit, I moved the context to the index page, so all the child components of the index will get access to the context: the currently selected language and a function to change the language (I put the function for the dropdown component to access TT)

tiramisuuuuuuu committed:
Created hover function and close when clicking out of div function

nandhanaselvam committed:
added basic structure of page links

nandhanaselvam committed:
added underline on hover. attempting bold on current page

nandhanaselvam committed:
fixed fonts

tiramisuuuuuuu committed:
Added some animation and styles

Made the dropdown rollout animation!

nandhanaselvam committed:
added image

nandhanaselvam committed:
attempting useState for highlight on active

tiramisuuuuuuu committed:
Added icons and dividers to dropdown

tiramisuuuuuuu committed:
Added more styling details!

tiramisuuuuuuu committed:
Finalized dropdown!! (added original language names and fixed context values)

Made the language names to be written in the corresponding language. Made the returned context values of each option to match the names of json's (en, zh, ko, vi, hmn, and es) :)

nandhanaselvam committed:
minor spacing fixes + added border on active page

nandhanaselvam committed:
Merge branch 'main' into feat/header-nandhana

nandhanaselvam committed:
added header folder for image

nandhanaselvam committed:
fixes- highlight, underline issue,

nandhanaselvam committed:
working on internationalization

JayJ104 committed:
Merge branch 'main' into feat/header-nandhana

nandhanaselvam committed:
trying to add internationalization

nandhanaselvam committed:
Merge branch 'feat/header-nandhana' of https://github.com/include-davis/paul-homm into feat/header-nandhana

nandhanaselvam committed:
Revert "Merge branch 'feat/header-nandhana' of https://github.com/include-davis/paul-homm into feat/header-nandhana"

This reverts commit 463725af7fe819680d2ea8b09ff6fe65e90765ff, reversing
changes made to 3c90ac6c758ee2ab7cf5363b54b2f781601c3a55.

tiramisuuuuuuu committed:
Added i18 features in dropdown

nandhanaselvam committed:
starting to implement mobile header

nandhanaselvam committed:
working on mobile design

nandhanaselvam committed:
added minor comment for clarity

tiramisuuuuuuu committed:
Added i18n compatibility + Condensed dropdown css & code

I deleted repetitive code in dropDown.jsx using object mapping & created a i18n.config. I also left an example, compatible to my additions in dropDown.jsx, in index.jsx on how I am imagining i18n would work

tiramisuuuuuuu committed:
Replaced imgs with React Icons + Cleaned up dropDown css

JayJ104 committed:
Merge branch 'main' into feat/lang_dropdown-sreya

JayJ104 committed:
Merge branch 'feat/lang_dropdown-sreya' of https://github.com/include-davis/paul-homm into feat/lang_dropdown-sreya

JayJ104 committed:
header link onclick fix

JayJ104 committed:
language change simplified

JayJ104 committed:
Merge branch 'main' into feat/header-nandhana

JayJ104 committed:
Merge branch 'main' into feat/header-nandhana

JayJ104 committed:
Update en.json

JayJ104 committed:
language dropdown positioning

JayJ104 committed:
Update footer.module.scss

JayJ104 commented on src/styles/components/header/header.module.scss:
```diff
@@ -66,17 +66,26 @@
+    border: none;
+    font-size: 4rem;
+    color: var(--color-gold);
+    padding-right: 20px;
```
no biggie but change it to 16px. tryna stick to multiples of 8px

JayJ104 commented on src/styles/components/header/header.module.scss:
```diff
@@ -7,7 +8,6 @@
     background-color: var(--color-maroon);
     height: 226px;
```
you can add 226px as a variable in globals.scss and then import it here as a var. actually name this one as desktop-header-height bc we'll need to define mobile-header-height too.

nandhanaselvam committed:
implemented tablet mixins

nandhanaselvam committed:
implemented mobile mixins

nandhanaselvam committed:
Merge branch 'main' into feat/header-nandhana

tiramisuuuuuuu committed:
Added mobile structure to dropdown

tiramisuuuuuuu committed:
Added header collapse/expand functionality

tiramisuuuuuuu committed:
Mobile dropdown toggle fixed

tiramisuuuuuuu committed:
Reorganized dropDown css file

JayJ104 reviewed (changes requested):
Looks great! Just need to fix the z-index of the hamburger menu so its not obstructed by the page contents and expand the clickable area of the navigation links vertically for user convenience.

nandhanaselvam committed:
increased z index of hamburger menu dropdown

sairaheta1 committed:
implemented fixes to link highliting

Corrects the errors where the links wouldn't span the entire menu. Only thing left is to fix the background on active pages to achieve the right dimensions & possibly add in a line transformation on hover.

nandhanaselvam committed:
fixed mobile dropdown (z index, highlights)

JayJ104 committed:
Merge branch 'main' into feat/header-nandhana

JayJ104 committed:
mobile dropdown simplified

JayJ104 committed:
Merge branch 'main' into feat/lang_dropdown-sreya

JayJ104 committed:
Merge branch 'feat/lang_dropdown-sreya' into feat/header-nandhana

JayJ104 committed:
language dropdown position fixed

JayJ104 committed:
Merge branch 'main' into feat/header-nandhana
